### ISSUE TITLE
PhpOffice\PhpSpreadsheet\Reader\Csv::inferSeparator() generate a notice with empty file

### ISSUE THREAD

**Issue by Raphhh** [PHPOffice/PhpSpreadsheet#337]
This is:

```
- [ x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?

IOFactory should load empty csv file.

### What is the current behavior?

IOFactory generates a notice when we load an empty csv file.

### What are the steps to reproduce?

```php
<?php
require __DIR__ . '/vendor/autoload.php';

$filepath = __DIR__ . '/empty.csv';
touch($filepath);

\PhpOffice\PhpSpreadsheet\IOFactory::load($filepath); //PHP Notice:  Undefined offset: -1 in vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Reader/Csv.php on line 185

```

### Which versions of PhpSpreadsheet and PHP are affected?

My env:
 - PHP:7.2.1-1+ubuntu14.04.1+deb.sury.org+1PHP 
 - phpoffice/phpspreadsheet:1.0.0

**Comment by PowerKiKi** [comment]
Would you submit a PR to fix that and cover it with unit tests ?

**Comment by floriandammeyer** [comment]
I encountered a similar issue when I tried to load an invalid CSV file for testing purposes using Laravel's `UploadedFile::fake()` method (meaning the file was not a CSV file at all but contained random bytes instead).

We have an upload form for CSV files, the uploaded files are read using `PhpOffice\PhpSpreadsheet\Reader\Csv`. To react to malformed files, our script relies on the `canRead()` method of the CSV reader. But that method only checks if the file exists, not if it is a CSV file that can be read. Therefore the script continues and runs into the same situation as stated above, a PHP Notice is emitted which is hard to react to. A `ReaderException` would be more helpful. 

Unfortunately I do not have the time to take a deeper look into this problem myself.

**Comment by PowerKiKi** [comment]
@floriandammeyer https://github.com/PHPOffice/PhpSpreadsheet/commit/e31878ceb12c3cb6deba8cbfbff9cd3f2de4de17 should help with your case.

**Comment by ricardosaracino** [comment]
im getting a separator ":" from a csv file

![image](https://user-images.githubusercontent.com/759253/39583348-e6cb1574-4ebd-11e8-845a-c1dd3c00cbd5.png)

**Comment by oleibman** [comment]
Fixed by PR #743.
